### ISSUE TITLE
Confirm Video deletion from local playlists

### ISSUE THREAD

**Issue by 5lttrz** [TeamNewPipe/NewPipe#7845]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### What feature do you want?

Confirming video deletion whenever a video is deleted from local playlists.

#### Why do you want this feature?

There might be a few videos whose thumbnail, link, and title are stored in the local playlist, but the video is not available. Accidental deletion can cause them to be lost forever with no way of revival. In fact thats why i made this feature request (lost a historic vid's link thats deleted)

**Comment by NewNewPipe** [comment]
Good Idea.

Confirmation of deletion is always useful, sometimes it saved from unwanted deletion.

**Comment by litetex** [comment]
I don't think this is necessary since we have the "restore" button when someone (accidentally) deletes a video.

**Comment by 5lttrz** [comment]
I dont have any undo or restore button when i delete a video.

**Comment by litetex** [comment]
> I don't have any undo or restore button when i delete a video.

Are you sure?
That's not possible because I have one:
<img src="https://user-images.githubusercontent.com/40789489/153649420-e4b2f97d-f302-4ae2-af90-5cd119216b8b.png" height=500 />
and the code also says that you should have one when deleting a video:
https://github.com/TeamNewPipe/NewPipe/blob/2e538b89599cfb6225221d22f1aca3e8eaa758ba/app/src/main/java/us/shandian/giga/ui/common/Deleter.java#L94

Are using NewPipe to delete files?

**Comment by SameenAhnaf** [comment]
@litetex The author is talking about playlist videos added by the user. Not downloads.

**Comment by XiangRongLin** [comment]
For consistency sake I would propose to have the same behaviour as deleting downloaded videos. So instantly delete it, but add a snackbar with the undo button

**Comment by XiangRongLin** [comment]
The code from the download section can't be used as is. Reason being that [Deleter.java](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/us/shandian/giga/ui/common/Deleter.java) is tightly coupled to the download classes (see constructor parameters). From a first glance I also can't see how to easily resolve this.

Main functionality of that class is (I think)
1. handling everything asynchonously
1. delaying the deletion until the snackbar disappears
2. preventing stacking snackbars when deleting multiple things

**Comment by samyberkane04** [comment]
https://github.com/TeamNewPipe/NewPipe/pull/8351 I try it to fix it in this PR .
